integration/hibernate-4.2/src/main/java/com/blazebit/persistence/integration/hibernate/Hibernate4Access.java

```diff
@@ -314,4 +331,40 @@ public QueryParameters createQueryParameters(
     public ParameterTranslations createParameterTranslations(List<ParameterSpecification> queryParameterSpecifications) {
         return new ParameterTranslationsImpl(queryParameterSpecifications);
     }
+
+    /**
+     * Iterator over {@code ScrollableResults}.
+     *
+     * @author Jan-Willem Gmelig Meyling
+     * @since 1.6.2
+     */
+    private static class ScrollableResultsIterator implements Iterator<Object> {
+        private final ScrollableResults scrollableResults;
+
+        public ScrollableResultsIterator(ScrollableResults scrollableResults) {
+            this.scrollableResults = scrollableResults;
+        }
+
+        @Override
+        public boolean hasNext() {
+            return scrollableResults.next();
```

Comment:
Again, implementation differs from the other implementations.

Reply:
Yes, because the call "needed" (I am questioning that in my other comment) was only introduced after 5.2.